geomesa-filter/src/main/scala/org/locationtech/geomesa/filter/expression/FastDWithin.scala

```diff
@@ -0,0 +1,64 @@
+/***********************************************************************
+ * Copyright (c) 2013-2018 Commonwealth Computer Research, Inc.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Apache License, Version 2.0
+ * which accompanies this distribution and is available at
+ * http://www.opensource.org/licenses/apache2.0.php.
+ ***********************************************************************/
+
+package org.locationtech.geomesa.filter.expression
+
+import com.vividsolutions.jts.geom.Geometry
+import com.vividsolutions.jts.operation.distance.DistanceOp
+import org.geotools.referencing.GeodeticCalculator
+import org.locationtech.geomesa.filter.FilterHelper
+import org.locationtech.geomesa.utils.geotools.GeometryUtils
+import org.opengis.filter.FilterVisitor
+import org.opengis.filter.MultiValuedFilter.MatchAction
+import org.opengis.filter.expression.{Expression, Literal}
+import org.opengis.filter.spatial.DWithin
+
+object FastDWithin {
+
+  private val calculators = new ThreadLocal[GeodeticCalculator] {
+    override def initialValue: GeodeticCalculator = new GeodeticCalculator()
```

Comment:
I think this duplicates the code/ideas of GeometricDistanceFunctions.fastDistance.

Reply:
is that a geotools thing?

Reply:
hmm, that doesn't seem to be fast haha, it just uses the geodetic calculator

Reply:
https://github.com/locationtech/geomesa/blob/master/geomesa-spark/geomesa-spark-sql/src/main/scala/org/locationtech/geomesa/spark/GeometricDistanceFunctions.scala